src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforceStreamingSource.java

```diff
@@ -0,0 +1,191 @@
+      }
+    } catch (ConnectionException e) {
+      throw new InvalidStageException("There was issue communicating with Salesforce", e);
+    }
+  }
+
+  @Override
+  public JavaDStream<StructuredRecord> getStream(StreamingContext streamingContext) throws ConnectionException {
+    config.validate(); // validate when macros are substituted
+    config.ensurePushTopicExistAndWithCorrectFields(); // run when macros are substituted
+
+    this.schema = streamingContext.getOutputSchema();
+
+    if (this.schema == null) { // if was not set in configurePipeline due to fields containing macro
+      this.schema = SalesforceSchemaUtil.getSchema(config.getAuthenticatorCredentials(),
+                                                   SObjectDescriptor.fromQuery(config.getPushTopicQuery()));
+    }
+    LOG.debug("Schema is {}", schema);
+
+    JavaStreamingContext jssc = streamingContext.getSparkStreamingContext();
+
+    return jssc.receiverStream(new SalesforceReceiver(this.config.getAuthenticatorCredentials(),
+                                                      this.config.getPushTopicName())).
+      map((Function<String, StructuredRecord>) this::getStructuredRecord).filter(Objects::nonNull);
+  }
+
+  private StructuredRecord getStructuredRecord(String jsonMessage) throws ConnectionException {
+    try {
+      StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+
+      JSONObject sObjectFields = new JSONObject(jsonMessage) // throws a JSONException if failed to decode
+        .getJSONObject("data") // throws a JSONException if not found
+        .getJSONObject("sobject");
+
+      for (Map.Entry<String, Object> entry : sObjectFields.toMap().entrySet()) {
+        String fieldName = entry.getKey();
+        Object value = entry.getValue();
+
+        Schema.Field field = schema.getField(fieldName, true);
+
+        if (field == null) {
+          continue; // this field is not in schema
+        }
+
+        builder.set(field.getName(), value);
+      }
+      return builder.build();
+    } catch (Exception ex) {
+      switch (config.getErrorHandling()) {
+        case SalesforceStreamingSourceConfig.ERROR_HANDLING_SKIP:
+          LOG.warn("Cannot process json '{}', skipping it.", jsonMessage, ex);
+          return null;
+        case SalesforceStreamingSourceConfig.ERROR_HANDLING_STOP:
+          throw ex;
+        default:
+          throw new UnexpectedFormatException(
+            String.format("Unknown error handling strategy '%s'", config.getErrorHandling()));
+      }
+    }
+  }
+
+  /**
+   * Request object for retrieving schema from the Salesforce
+   */
+  class Request {
+    String username;
+    String password;
+    String clientId;
+    String clientSecret;
+    String loginUrl;
+
+    String pushTopicName;
+    String pushTopicQuery;
+    String sObjectName;
+  }
+
+  @Path("outputSchema")
+  public Schema outputSchema(SalesforceStreamingSource.Request request) throws Exception {
```

Comment:
Why not use the same approach as in Batch Source? You won't need to create Request object: https://github.com/data-integrations/salesforce/blob/master/src/main/java/co/cask/hydrator/salesforce/plugin/source/batch/SalesforceBatchSource.java#L174